File: app/src/main/java/com/emiryanvl/tapreader/data/local/dataSources/LocalBookDataSource.java
package com.emiryanvl.tapreader.data.local.dataSources;

import com.emiryanvl.tapreader.domain.model.Book;

import java.util.List;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Observable;

public interface LocalBookDataSource {

    Observable<List<Book>> getAllBooks();

    Observable<Book> getBook(int id);

    Completable addBook(Book book);

    void updateBook(int id, Book book);

    void deleteBook(int id);
}

File: app/src/main/java/com/emiryanvl/tapreader/data/local/dataSources/LocalBookDataSource.kt
package com.emiryanvl.tapreader.data.local.dataSources

import com.emiryanvl.tapreader.domain.models.Book
import kotlinx.coroutines.flow.Flow

interface LocalBookDataSource {

    fun getAllBooks(): Flow<List<Book>>

    fun getBook(id: Int): Flow<Book>

    suspend fun addBook(book: Book)

    suspend fun updateBook(id: Int, book: Book)

    suspend fun deleteBook(id: Int)
}